ResearchKit/Common/ORKFormStepViewController.m

```diff
@@ -891,7 +900,12 @@ - (void)tableView:(UITableView *)tableView didSelectRowAtIndexPath:(NSIndexPath
         ORKTableCellItem *cellItem = section.items[indexPath.row - 1];
         [section.textChoiceCellGroup didSelectCellAtIndexPath:indexPath];
         
+        // formItem's questionType is ORKQuestionTypeSingleChoice even if answerFormat is ORKAnswerFormatBoolean
+        // should check for answerFormat rather than questionType
+        /*
         id answer = (cellItem.formItem.questionType == ORKQuestionTypeBoolean)? [section.textChoiceCellGroup answerForBoolean] : [section.textChoiceCellGroup answer];
+        */
+        id answer = ([cellItem.formItem.answerFormat isKindOfClass:[ORKBooleanAnswerFormat class]])? [section.textChoiceCellGroup answerForBoolean] : [section.textChoiceCellGroup answer];
```

Comment:
questionType won't change even `ORKBooleanAnswerFormat` use `ORKChoiceAnswerFormat` as implied answer format. So this change would make no difference.